src/components/shared/Modal.css

```diff
@@ -0,0 +1,42 @@
+.modal-wrapper {
+  position: fixed;
+  display: flex;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  z-index: -1;
+  top: 0;
+  left: 0;
+  transition: z-index 0.4s ease-in-out;
```

Comment:
I added this slightly longer transition to the z-index change to make sure we have enough time for the animation to play.